src/TcpConnector.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace React\SocketClient;
+
+use React\EventLoop\LoopInterface;
+use React\Dns\Resolver\Resolver;
+use React\Stream\Stream;
+use React\Promise;
+use React\Promise\Deferred;
+
+class TcpConnector implements ConnectorInterface
+{
+    private $loop;
+
+    public function __construct(LoopInterface $loop)
+    {
+        $this->loop = $loop;
+    }
+
+    public function create($ip, $port)
+    {
+        if (false === filter_var($ip, FILTER_VALIDATE_IP)) {
+            return Promise\reject(new \InvalidArgumentException('Given parameter "' . $ip . '" is not a valid IP'));
+        }
+
+        $url = $this->getSocketUrl($ip, $port);
+
+        $socket = stream_socket_client($url, $errno, $errstr, 0, STREAM_CLIENT_CONNECT | STREAM_CLIENT_ASYNC_CONNECT);
+
+        if (!$socket) {
+            return Promise\reject(new \RuntimeException(
+                sprintf("connection to %s:%d failed: %s", $ip, $port, $errstr),
```

Comment:
Thanks for spotting, fixed! :)

For the reference: This code has been like this ever since the very first commit in this repo.